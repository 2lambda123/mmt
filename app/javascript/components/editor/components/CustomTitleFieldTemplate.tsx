/* eslint-disable react/static-property-placement */
/* eslint-disable react/no-unused-prop-types */
import { observer } from 'mobx-react'
import React from 'react'
import MetadataEditor from '../MetadataEditor'

type CustomTitleFieldTemplateProps = {
    title: string,
    required: boolean,
    schema: unknown,
    options: {
      editor: MetadataEditor
    }
  }

class CustomTitleFieldTemplate extends React.Component <CustomTitleFieldTemplateProps, never> {
  render() {
    const {
      title, required
    } = this.props
    return (
      <div>
        <div style={{
          borderBottom: 'solid 2px rgb(240,240,240', paddingBottom: 6, marginTop: 8, marginBottom: 8
        }}
        >
          <span className="title">
            {title}
          </span>
          {required ? <i className="eui-icon eui-required-o" style={{ color: 'green', padding: '5px' }} /> : ''}
        </div>
      </div>
    )
  }
}

export default observer(CustomTitleFieldTemplate)
