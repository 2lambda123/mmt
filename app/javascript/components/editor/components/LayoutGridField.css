.mediumTitle {
    font-size: 20px;
    font-weight: 550;
}

.col-children {
    border-Left: solid 5px rgb(240, 240, 240);
    padding-left: 15px
}

.row-children {
    margin-bottom: -5px
}

.group-description {
    padding-bottom: 30px;
}

.row-fieldset {
    margin-bottom: -5px;
}

.field-left-border {
    border-left: solid 5px rgb(240,240,240);
    margin-left: 10px;
}
.field-left-border-padded {
    border-left: solid 5px rgb(240,240,240);
    margin-left: 20px;
}
.field-bottom-border {
    padding-bottom: 10px;
    border-bottom: solid 2px rgb(240,240,240);
}