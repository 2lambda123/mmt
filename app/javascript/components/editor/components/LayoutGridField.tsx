import React, { FunctionComponent } from 'react'
import ObjectField, { ObjectFieldProps } from 'react-jsonschema-form/lib/components/fields/ObjectField'
import { retrieveSchema } from 'react-jsonschema-form/lib/utils'
import { Col } from 'react-bootstrap'
import { JSONSchema6 } from 'json-schema'
import { kebabCase } from 'lodash'

type ComponentProps = {
  [name: string]: string | JSONSchema6 | object;
};

type LayoutGridSchemaProps = {
  [Key: string]: any
}

/**
 * Custom field component for react-jsonschema-form which provides support for bootstrap grid system.
 * Code is based off:
 * https://github.com/audibene-labs/react-jsonschema-form-layout/blob/master/src/index.js
 */
export default class LayoutGridField extends ObjectField<ObjectFieldProps, never> {
  render() {
    const { uiSchema } = this.props
    let { layoutGridSchema } = this.props
    if (!layoutGridSchema) layoutGridSchema = uiSchema['ui:layout_grid']

    if (layoutGridSchema['ui:row']) {
      return this.renderRow(layoutGridSchema)
    } if (layoutGridSchema['ui:col']) {
      return this.renderCol(layoutGridSchema)
    }
    return this.renderField(layoutGridSchema)
  }

  renderRow(layoutGridSchema: LayoutGridSchemaProps) {
    const rows = layoutGridSchema['ui:row']
    const group = layoutGridSchema['ui:group']
    const groupDescription = layoutGridSchema['ui:group-description']

    if (group) {
      const { fields, formContext } = this.props.registry
      const { TitleField } = fields
      const { required, schema } = this.props
      const { description = '' } = schema

      const title = group && typeof group === 'string' ? group : null
      return (
        <fieldset
          className="rjsf-layout-grid-group"
        >
          <span data-testid={`layout-grid-field__row-title-field--${kebabCase(title)}`}>
            {title ? (
              <TitleField
                title={title}
                required={required}
                formContext={formContext}
              />
            ) : null}
          </span>
          { groupDescription ? (
            <div style={{ paddingBottom: 30 }}>
              { description }
            </div>
          ) : null }
          <div className="row" key={JSON.stringify(rows)}>{this.renderChildren(rows)}</div>
        </fieldset>
      )
    }
    return <div className="row" key={JSON.stringify(rows)}>{this.renderChildren(rows)}</div>
  }

  renderCol(layoutGridSchema: LayoutGridSchemaProps) {
    const { children, ...colProps } = layoutGridSchema['ui:col']

    const group = layoutGridSchema['ui:group']
    const groupDescription = layoutGridSchema['ui:group-description']
    if (group) {
      const { fields, formContext } = this.props.registry
      const { TitleField } = fields
      const { required, schema } = this.props
      const { description = '' } = schema
      const title = group && typeof group === 'string' ? group : null

      return (
        <Col {...colProps} key={JSON.stringify(colProps)}>
          <fieldset
            className="rjsf-layout-grid-group"
          >
            <span data-testid={`layout-grid-field__col-title-field--${kebabCase(title)}`}>
              {title ? (
                <TitleField
                  title={title}
                  required={required}
                  formContext={formContext}
                />
              ) : null}
            </span>
            { groupDescription ? (
              <div style={{ paddingBottom: 30 }}>
                { description }
              </div>
            ) : null }
            {this.renderChildren(children)}
          </fieldset>
        </Col>
      )
    }
    const {
      type = '',
      keywordScheme,
      keywordSchemeColumnNames = [],
      keywords,
      ...rest
    } = colProps
    return <Col {...colProps} key={JSON.stringify(colProps)}>{this.renderChildren(children)}</Col>
  }

  renderChildren(childrenLayoutGridSchema: LayoutGridSchemaProps) {
    const { definitions } = this.props.registry
    const schema = retrieveSchema(this.props.schema, definitions)

    return childrenLayoutGridSchema.map((layoutGridSchema: LayoutGridSchemaProps) => (
      <LayoutGridField
        {...this.props}
        key={JSON.stringify(layoutGridSchema)}
        schema={schema}
        layoutGridSchema={layoutGridSchema}
      />
    ))
  }

  renderField(layoutGridSchema: LayoutGridSchemaProps) {
    const {
      uiSchema,
      errorSchema,
      idSchema,
      disabled,
      readonly,
      onBlur,
      onFocus,
      formData
    } = this.props
    const { definitions, fields } = this.props.registry
    const { SchemaField } = fields
    const schema = retrieveSchema(this.props.schema, definitions)
    let name
    let render
    if (typeof layoutGridSchema === 'string') {
      name = layoutGridSchema
    } else {
      name = layoutGridSchema.name
      render = layoutGridSchema.render
    }
    if (schema.properties[name] && !render) {
      return (
        <span data-testid={`layout-grid-field__schema-field--${kebabCase(name)}`}>
          <SchemaField
            key={JSON.stringify(name)}
            name={name}
            required={this.isRequired(name)}
            schema={schema.properties[name]}
            uiSchema={uiSchema[name]}
            errorSchema={errorSchema[name]}
            idSchema={idSchema[name]}
            formData={formData[name]}
            onChange={this.onPropertyChange(name)}
            onBlur={onBlur}
            onFocus={onFocus}
            registry={this.props.registry}
            disabled={disabled}
            readonly={readonly}
            onKeyChange={this.onKeyChange(name)}
          />
        </span>
      )
    }
    let UIComponent: FunctionComponent<ComponentProps> = render

    UIComponent = render

    return (
      <span data-testid={`layout-grid-field__schema-field--${kebabCase(name)}`}>
        <UIComponent
          {...this.props}
          name={name}
          formData={formData}
          errorSchema={errorSchema}
          uiSchema={uiSchema}
          schema={schema}
          registry={this.props.registry}
        />
      </span>
    )
  }
}
