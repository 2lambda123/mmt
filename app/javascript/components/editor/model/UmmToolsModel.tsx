/* eslint-disable @typescript-eslint/no-explicit-any */
import toolInformationUiSchema from '../data/configuration/uiSchemas/tools/tool_information'
import relatedUrlsUiSchema from '../data/configuration/uiSchemas/tools/related_urls'
import toolsConfiguration from '../data/configuration/uiForms/toolsConfiguration'
import organizationUiSchema from '../data/configuration/uiSchemas/tools/organization'
import UmmModel from './UmmModel'
import descriptiveKeywordsUiSchema from '../data/configuration/uiSchemas/tools/descriptive_keywords'
import potentialActionUiSchema from '../data/configuration/uiSchemas/tools/potential_action'

export default class UmmToolsModel extends UmmModel {
  get documentTypeForDisplay() {
    return 'Tool Record'
  }
  get documentType() {
    return 'tool_drafts'
  }
  get formSections() {
    return toolsConfiguration
  }

  get uiSchema() {
    const base = { 'ui:submitButtonOptions': { norender: true, submitText: 'Save' } }

    if (this.currentSection.displayName === 'Tool Information') {
      const uiSchema: any = toolInformationUiSchema
      const urlUiSchema: any = uiSchema.URL
      urlUiSchema['ui:service'] = this.service
      urlUiSchema['ui:keyword_scheme'] = 'rucontenttype'
      return { ...uiSchema, ...base }
    }
    if (this.currentSection.displayName === 'Descriptive Keywords') {
      const uiSchema = descriptiveKeywordsUiSchema
      const toolKeywords: any = uiSchema.ToolKeywords
      toolKeywords['ui:service'] = this.service
      toolKeywords['ui:keyword_scheme'] = 'science_keywords'
      return uiSchema
    }
    if (this.currentSection.displayName === 'Tool Organizations') {
      const uiSchema: any = organizationUiSchema
      const organizations: any = uiSchema.Organizations.items
      organizations['ui:service'] = this.service
      organizations['ui:keyword_scheme'] = 'providers'
      organizations['ui:keyword_scheme_column_names'] = ['short_name', 'long_name', 'url']
      return { ...uiSchema, ...base }
    }
    if (this.currentSection.displayName === 'Potential Action') {
      const uiSchema: any = potentialActionUiSchema
      return { ...uiSchema, ...base }
    }
    if (this.currentSection.displayName === 'Related URLs') {
      const uiSchema: any = relatedUrlsUiSchema
      const urlUiSchema: any = uiSchema.RelatedURLs.items
      urlUiSchema['ui:service'] = this.service
      return { ...uiSchema, ...base }
    }
    return { ...base }
  }

  // In this case the old preview sends the same section name that the react component uses (just lowercase), so
  // nothing needs to be migrated. once the old MMT preview is replaced, this 'migrate" functionality can be removed.
  migratedSectionName(sectionName: string): string {
    return sectionName
  }
}
