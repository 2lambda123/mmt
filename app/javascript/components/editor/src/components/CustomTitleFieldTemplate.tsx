/* eslint-disable react/static-property-placement */
/* eslint-disable react/no-unused-prop-types */
import { FieldProps } from '@rjsf/utils'
import { observer } from 'mobx-react'
import React from 'react'
import './CustomTitleFieldTemplate.css'

interface CustomTitleFieldTemplateProps extends FieldProps {
  title: string,
  required: boolean,
  schema: unknown,
}

class CustomTitleFieldTemplate extends React.Component<CustomTitleFieldTemplateProps, never> {
  private scrollRef: React.RefObject<HTMLDivElement>

  constructor(props: CustomTitleFieldTemplateProps) {
    super(props)
    this.scrollRef = React.createRef()
  }

  executeScroll = () => this.scrollRef.current?.scrollIntoView({ behavior: 'smooth' })

  render() {
    const {
      title, required, registry, className = 'custom-title'
    } = this.props
    const { formContext } = registry
    const { editor } = formContext
    const { focusField } = editor
    if (title.replace(' ', '') === focusField) {
      setTimeout(() => {
        this.executeScroll()
      })
    }

    return (
      <div>
        <div ref={this.scrollRef} className="custom-title-header">
          <span className={className}>
            {title}
            {required ? <i className="eui-icon eui-required-o required-icon" /> : ''}
          </span>
        </div>
      </div>
    )
  }
}

export default observer(CustomTitleFieldTemplate)
