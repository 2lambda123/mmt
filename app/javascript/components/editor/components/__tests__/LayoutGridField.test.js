import React from 'react'
import {
  render, screen
} from '@testing-library/react'
import Form from '@rjsf/bootstrap-4'
import validator from '@rjsf/validator-ajv8'
import LayoutGridField from '../LayoutGridField'
import CustomTitleFieldTemplate from '../CustomTitleFieldTemplate'
import MetadataEditor from '../../MetadataEditor'
import UmmToolsModel from '../../model/UmmToolsModel'

const schema = {
  title: 'Todo',
  type: 'object',
  required: ['title'],
  properties: {
    password: {
      type: 'string',
      title: 'Password'
    },
    lastName: {
      type: 'string',
      title: 'Last name'
    },
    bio: {
      type: 'string',
      title: 'Bio'
    },
    firstName: {
      type: 'string',
      title: 'First name'
    },
    age: {
      type: 'integer',
      title: 'Age'
    }
  }
}
const fields = {
  layout_grid: LayoutGridField,
  TitleField: CustomTitleFieldTemplate
}
const model = new UmmToolsModel()
const editor = new MetadataEditor(model)
LayoutGridField.defaultProps = { options: { editor } }
describe('Layout Grid Field Component', () => {
  it('renders a basic form', async () => {
    const uiSchema = {
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            'ui:col': {
              md: 12,
              children: [
                {
                  'ui:row': [
                    { 'ui:col': { md: 6, children: ['firstName'] } },
                    { 'ui:col': { md: 6, children: ['lastName'] } }
                  ]
                },
                {
                  'ui:row': [
                    { 'ui:col': { md: 6, children: ['password'] } },
                    { 'ui:col': { md: 6, children: ['age'] } }
                  ]
                },
                {
                  'ui:row': [
                    { 'ui:col': { md: 12, children: ['bio'] } }
                  ]
                }
              ]
            }
          }
        ]
      }
    }

    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} fields={fields} />)
    expect(screen.getByTestId('layout-grid-field__schema-field--first-name')).toHaveTextContent('First name')
    expect(screen.getByTestId('layout-grid-field__schema-field--last-name')).toHaveTextContent('Last name')
    expect(screen.getByTestId('layout-grid-field__schema-field--password')).toHaveTextContent('Password')
    expect(screen.getByTestId('layout-grid-field__schema-field--age')).toHaveTextContent('Age')
    expect(screen.getByTestId('layout-grid-field__schema-field--bio')).toHaveTextContent('Bio')
    expect(container).toMatchSnapshot()
  })

  it('renders a group row title', async () => {
    const uiSchema = {
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            'ui:col': {
              md: 12,
              children: [
                {
                  'ui:group': 'Full Name',
                  'ui:row': [
                    { 'ui:col': { md: 6, children: ['firstName'] } },
                    { 'ui:col': { md: 6, children: ['lastName'] } }
                  ]
                }]
            }
          }
        ]
      }
    }
    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} fields={fields} />)
    expect(screen.getByTestId('layout-grid-field__schema-field--first-name')).toHaveTextContent('First name')
    expect(screen.getByTestId('layout-grid-field__schema-field--last-name')).toHaveTextContent('Last name')
    expect(screen.getByTestId('layout-grid-field__row-title-field--full-name')).toHaveTextContent('Full Name')
    expect(container).toMatchSnapshot()
  })

  it('renders a group row title which is undefined', async () => {
    const uiSchema = {
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            'ui:col': {
              md: 12,
              children: [
                {
                  'ui:group': {},
                  'ui:row': [
                    { 'ui:col': { md: 6, children: ['firstName'] } },
                    { 'ui:col': { md: 6, children: ['lastName'] } }
                  ]
                }]
            }
          }
        ]
      }
    }
    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} fields={fields} />)
    expect(screen.getByTestId('layout-grid-field__schema-field--first-name')).toHaveTextContent('First name')
    expect(screen.getByTestId('layout-grid-field__schema-field--last-name')).toHaveTextContent('Last name')
    expect(screen.queryByTestId('layout-grid-field__schema-field--full-name')).toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('renders a group column title', async () => {
    const uiSchema = {
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            'ui:group': 'Biography',
            'ui:col': { md: 12, children: ['bio'] }
          }
        ]
      }
    }
    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} fields={fields} />)
    expect(screen.getByTestId('layout-grid-field__col-title-field--biography')).toHaveTextContent('Biography')
    expect(container).toMatchSnapshot()
  })

  it('renders a group column title which undefined', async () => {
    const uiSchema = {
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            'ui:group': {},
            'ui:col': { md: 12, children: ['bio'] }
          }
        ]
      }
    }
    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} fields={fields} />)
    expect(screen.queryByTestId('layout-grid-field__col-title-field--biography')).toBeNull()
    expect(container).toMatchSnapshot()
  })

  it('renders a custom field', async () => {
    const uiSchema = {
      'ui:field': 'layout_grid',
      'ui:layout_grid': {
        'ui:row': [
          {
            name: 'password',
            render: () => (
              <div>My Custom Component</div>
            )
          },
          { name: 'age' }
        ]
      }
    }
    const { container } = render(<Form validator={validator} schema={schema} uiSchema={uiSchema} fields={fields} />)
    expect(screen.getByTestId('layout-grid-field__schema-field--password')).toHaveTextContent('My Custom Component')
    expect(container).toMatchSnapshot()
  })
})
