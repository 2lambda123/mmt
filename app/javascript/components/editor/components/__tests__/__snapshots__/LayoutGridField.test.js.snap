// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Grid Field Component renders a basic form 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <span
                data-testid="layout-grid-field__schema-field--first-name"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="mb-0 form-group"
                  >
                    <label
                      class="form-label"
                    >
                      First name
                    </label>
                    <input
                      class="form-control"
                      id="root_firstName"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </span>
            </div>
            <div
              class="col-md-6"
            >
              <span
                data-testid="layout-grid-field__schema-field--last-name"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="mb-0 form-group"
                  >
                    <label
                      class="form-label"
                    >
                      Last name
                    </label>
                    <input
                      class="form-control"
                      id="root_lastName"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <span
                data-testid="layout-grid-field__schema-field--password"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="mb-0 form-group"
                  >
                    <label
                      class="form-label"
                    >
                      Password
                    </label>
                    <input
                      class="form-control"
                      id="root_password"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </span>
            </div>
            <div
              class="col-md-6"
            >
              <span
                data-testid="layout-grid-field__schema-field--age"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="mb-0 form-group"
                  >
                    <label
                      class="form-label"
                    >
                      Age
                    </label>
                    <input
                      class="form-control"
                      id="root_age"
                      placeholder=""
                      type="integer"
                      value=""
                    />
                  </div>
                </div>
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-md-12"
            >
              <span
                data-testid="layout-grid-field__schema-field--bio"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="mb-0 form-group"
                  >
                    <label
                      class="form-label"
                    >
                      Bio
                    </label>
                    <input
                      class="form-control"
                      id="root_bio"
                      placeholder=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Layout Grid Field Component renders a custom field 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group"
    >
      <div
        class="row"
      >
        <span
          data-testid="layout-grid-field__schema-field--password"
        >
          <div>
            My Custom Component
          </div>
        </span>
        <span
          data-testid="layout-grid-field__schema-field--age"
        >
          <div
            class="form-group"
          >
            <div
              class="mb-0 form-group"
            >
              <label
                class="form-label"
              >
                Age
              </label>
              <input
                class="form-control"
                id="root_age"
                placeholder=""
                type="integer"
                value=""
              />
            </div>
          </div>
        </span>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Layout Grid Field Component renders a group column title 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <fieldset
            class="rjsf-layout-grid-group"
          >
            <span
              data-testid="layout-grid-field__col-title-field--biography"
            >
              <div
                class="my-1"
              >
                <h5>
                  Biography
                </h5>
                <hr
                  class="border-0 bg-secondary"
                  style="height: 1px;"
                />
              </div>
            </span>
            <span
              data-testid="layout-grid-field__schema-field--bio"
            >
              <div
                class="form-group"
              >
                <div
                  class="mb-0 form-group"
                >
                  <label
                    class="form-label"
                  >
                    Bio
                  </label>
                  <input
                    class="form-control"
                    id="root_bio"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </span>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Layout Grid Field Component renders a group column title which undefined 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <fieldset
            class="rjsf-layout-grid-group"
          >
            <span
              data-testid="layout-grid-field__col-title-field--"
            />
            <span
              data-testid="layout-grid-field__schema-field--bio"
            >
              <div
                class="form-group"
              >
                <div
                  class="mb-0 form-group"
                >
                  <label
                    class="form-label"
                  >
                    Bio
                  </label>
                  <input
                    class="form-control"
                    id="root_bio"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </span>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Layout Grid Field Component renders a group row title 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <fieldset
            class="rjsf-layout-grid-group"
          >
            <span
              data-testid="layout-grid-field__row-title-field--full-name"
            >
              <div
                class="my-1"
              >
                <h5>
                  Full Name
                </h5>
                <hr
                  class="border-0 bg-secondary"
                  style="height: 1px;"
                />
              </div>
            </span>
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <span
                  data-testid="layout-grid-field__schema-field--first-name"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="mb-0 form-group"
                    >
                      <label
                        class="form-label"
                      >
                        First name
                      </label>
                      <input
                        class="form-control"
                        id="root_firstName"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </span>
              </div>
              <div
                class="col-md-6"
              >
                <span
                  data-testid="layout-grid-field__schema-field--last-name"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="mb-0 form-group"
                    >
                      <label
                        class="form-label"
                      >
                        Last name
                      </label>
                      <input
                        class="form-control"
                        id="root_lastName"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </span>
              </div>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`Layout Grid Field Component renders a group row title which is undefined 1`] = `
<div>
  <form
    class="rjsf"
  >
    <div
      class="form-group"
    >
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <fieldset
            class="rjsf-layout-grid-group"
          >
            <span
              data-testid="layout-grid-field__row-title-field--"
            />
            <div
              class="row"
            >
              <div
                class="col-md-6"
              >
                <span
                  data-testid="layout-grid-field__schema-field--first-name"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="mb-0 form-group"
                    >
                      <label
                        class="form-label"
                      >
                        First name
                      </label>
                      <input
                        class="form-control"
                        id="root_firstName"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </span>
              </div>
              <div
                class="col-md-6"
              >
                <span
                  data-testid="layout-grid-field__schema-field--last-name"
                >
                  <div
                    class="form-group"
                  >
                    <div
                      class="mb-0 form-group"
                    >
                      <label
                        class="form-label"
                      >
                        Last name
                      </label>
                      <input
                        class="form-control"
                        id="root_lastName"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </span>
              </div>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
